core/services/vrf/vrf_multiword_consumer_test.go

```diff
@@ -0,0 +1,70 @@
+package vrf_test
+
+import (
+	"fmt"
+	"math/big"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/smartcontractkit/chainlink/core/internal/cltest"
+	"github.com/smartcontractkit/chainlink/core/services/vrf"
+	"github.com/smartcontractkit/chainlink/core/store/models"
+	"github.com/stretchr/testify/require"
+)
+
+func TestMultiwordVRFConsumer(t *testing.T) {
+	key := cltest.MustGenerateRandomKey(t)
+	coordinator := newVRFCoordinatorUniverse(t, key)
+	contract := coordinator.multiwordConsumerContract
+	keyHash, _, fee := registerProvingKey(t, coordinator)
+	numwords := 50
+	bigNumwords := big.NewInt(int64(numwords))
+	seed := 0
+	_, err := contract.TestRequestRandomness(
+		coordinator.carol, keyHash, fee, big.NewInt(int64(seed)), bigNumwords)
+	require.NoError(t, err, "could not send randomness request")
+	coordinator.backend.Commit()
+
+	rawLog, err := coordinator.rootContract.FilterRandomnessRequest(nil, nil)
+	require.NoError(t, err, "failed to subscribe to RandomnessRequest logs")
+	logCount := 0
+	for rawLog.Next() {
+		logCount += 1
+	}
+	require.Equal(t, 1, logCount, "unexpected log generated by randomness request to VRFCoordinator")
+	log := models.RawRandomnessRequestLogToRandomnessRequestLog(
+		(*models.RawRandomnessRequestLog)(rawLog.Event))
+
+	preSeed, err := vrf.BigToSeed(log.Seed)
+	require.NoError(t, err, "pre-seed %x out of range", preSeed)
+	s := vrf.PreSeedData{
+		PreSeed:   preSeed,
+		BlockHash: log.Raw.Raw.BlockHash,
+		BlockNum:  log.Raw.Raw.BlockNumber,
+	}
+	proofBlob, err := vrf.GenerateProofResponseWithNonce(rawSecretKey, s, big.NewInt(1) /* nonce */)
+	require.NoError(t, err, "could not generate VRF proof!")
+	// Seems to be a bug in the simulated backend: without this extra Commit, the
+	// EVM seems to think it's still on the block in which the request was made,
+	// which means that the relevant blockhash is unavailable.
+	coordinator.backend.Commit()
+
+	// Seems to be a bug in the simulated backend: without this extra Commit, the
```

Comment:
Intentional duplicate of comment + `.Commit()`?

Reply:
Thanks for the feedback. This was written in a hurry the other day, due to an immediate need. We're waiting to hear from product about whether this or @kaleofduty's solution is better. If they want to go with this, I'll clean the test up. Meanwhile, I'll put this PR in draft state.